## Task 1

In [23]:
import pandas as pd
import numpy as np
print(pd.__version__)

2.3.3


## Task2

In [ ]:
customers = pd.read_csv("data/customers.csv")
customers

## Task3

In [ ]:
customers.head(6)

In [ ]:
customers.tail(4)

In [ ]:
customers.sample(5)

## Task4

In [ ]:
cars = pd.read_csv("data/cars.csv")
cars.info()


- 398 Rows
- 9 Columns
- MPG, Cylinders, Displacement, Horsepower, Weight, Acceleration, Model Year, Origin, Car Name

## Task5
- Whole Numbers: Cylinders, Weight, Model Year, Origin
- Floats: MPG, Displacement, Acceleration
- Objects: Horsepower, Car Name

## Task6

In [20]:
titanic = pd.read_csv("data/titanic_survival.csv")
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   survived     891 non-null    int64  
 1   pclass       891 non-null    int64  
 2   sex          891 non-null    object 
 3   age          714 non-null    float64
 4   sibsp        891 non-null    int64  
 5   parch        891 non-null    int64  
 6   fare         891 non-null    float64
 7   embarked     889 non-null    object 
 8   class        891 non-null    object 
 9   who          891 non-null    object 
 10  adult_male   891 non-null    bool   
 11  deck         203 non-null    object 
 12  embark_town  889 non-null    object 
 13  alive        891 non-null    object 
 14  alone        891 non-null    bool   
dtypes: bool(2), float64(2), int64(4), object(7)
memory usage: 92.4+ KB


- 891 Rows, 15 columns
- Most missing values are in "deck"
- You know a value is missing by the "Non-Null Count" column, where the difference between the non-null count and the index range is the amount of missing values.

## Task7

In [21]:
cars.describe()

,mpg,cylinders,displacement,weight,acceleration,model year,origin
count,398.000000,398.000000,398.000000,398.000000,398.000000,398.000000,398.000000
mean,23.514573,5.454774,193.425879,2970.424623,15.568090,76.010050,1.572864
std,7.815984,1.701004,104.269838,846.841774,2.757689,3.697627,0.802055
min,9.000000,3.000000,68.000000,1613.000000,8.000000,70.000000,1.000000
25%,17.500000,4.000000,104.250000,2223.750000,13.825000,73.000000,1.000000
50%,23.000000,4.000000,148.500000,2803.500000,15.500000,76.000000,1.000000
75%,29.000000,8.000000,262.000000,3608.000000,17.175000,79.000000,2.000000
max,46.600000,8.000000,455.000000,5140.000000,24.800000,82.000000,3.000000


- 23.51
- Displacement: 455.0
- Median Weight: 2803

## Task 8

In [28]:
print(titanic["fare"].mean())
print(titanic["fare"].median())
print(titanic["fare"].std())
print(titanic["fare"].min())
print(titanic["fare"].max())

32.204207968574636
14.4542
49.693428597180905
0.0
512.3292
